### PR TITLE
build: OwlBot config to specify correct paths with module names

### PR DESCRIPTION
**Pull request by suztomo**
Towards https://github.com/googleapis/google-cloud-java/issues/8479

- build: set_owlbot_config
- Updating OwlBot configuration files with modules

### CODE REVIEW

File: java-bigquerydatatransfer/.OwlBot.yaml [hunk 1]
```diff
@@ -14,57 +14,57 @@
...
-- "/proto-google-.*/src"
-- "/google-.*/src"
+- "/java-bigquerydatatransfer/grpc-google-.*/src"
+- "/java-bigquerydatatransfer/proto-google-.*/src"
+- "/java-bigquerydatatransfer/google-.*/src"
 
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-bigquerydatatransfer/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
 - "/google-cloud-bigquerydatatransfer/src/main/java/com/google/cloud/bigquery/datatransfer/v1/SampleApp.java"
```
Comment (suztomo) [inline java-bigquerydatatransfer/.OwlBot.yaml:23]: This is incorrect.

File: java-kms/.OwlBot.yaml [hunk 2]
```diff
@@ -14,18 +14,18 @@
...
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
-- "/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/UntypedKeyName.java"
-- "/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/KeyName.java"
-- "/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/KeyNames.java"
+- "/java-kms/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-kms/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/UntypedKeyName.java"
+- "/java-kms/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/KeyName.java"
+- "/java-kms/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/KeyNames.java"
 - "/google-cloud-kms/src/test/java/com/google/cloud/kms/it/ITKmsTest.java"
```
Comment (suztomo) [inline java-kms/.OwlBot.yaml:27]: This is missed.

File: java-secretmanager/.OwlBot.yaml [hunk 3]
```diff
@@ -14,18 +14,18 @@
...
 
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
-- "/proto-google-cloud-secretmanager-v1beta1"
-- "/grpc-google-cloud-secretmanager-v1beta1"
+- "/java-secretmanager/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-secretmanager/proto-google-cloud-secretmanager-v1beta1"
+- "/java-secretmanager/grpc-google-cloud-secretmanager-v1beta1"
 - "/google-cloud-secretmanager/src/main/java/com/google/cloud/secretmanager/v1beta1"
 - "/google-cloud-secretmanager/src/test/java/com/google/cloud/secretmanager/v1beta1"
```
Comment (suztomo) [inline java-secretmanager/.OwlBot.yaml:27]: This is missed

File: java-trace/.OwlBot.yaml [hunk 4]
```diff
@@ -14,12 +14,12 @@
...
-- "/google-.*/src"
+- "/java-trace/grpc-google-.*/src"
+- "/java-trace/proto-google-.*/src"
+- "/java-trace/google-.*/src"
 
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-trace/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
 - "/google-cloud-trace/src/test/java/com/google/cloud/trace/v1/VPCServiceControlTest.java"
 - "/google-cloud-trace/src/test/java/com/google/cloud/trace/v2/VPCServiceControlTest.java"
```
Comment (suztomo) [inline java-trace/.OwlBot.yaml:24]: These are missed.

File: java-vision/.OwlBot.yaml [hunk 5]
```diff
@@ -14,14 +14,14 @@
...
+- "/java-vision/google-.*/src"
 
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
-- "/proto-google-cloud-vision-v1/src/main/java/com/google/cloud/vision/v1/ImageName.java"
+- "/java-vision/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-vision/proto-google-cloud-vision-v1/src/main/java/com/google/cloud/vision/v1/ImageName.java"
 - "/google-cloud-vision/src/test/java/com/google/cloud/vision/it/ITSystemTest.java"
 - "/google-cloud-vision/src/test/resources/city.jpg"
 - "/google-cloud-vision/src/test/resources/face_no_surprise.jpg"
```
Comment (suztomo) [inline java-vision/.OwlBot.yaml:27]: These are missed

File: java-websecurityscanner/.OwlBot.yaml [hunk 6]
```diff
@@ -14,13 +14,13 @@
...
+- "/java-websecurityscanner/samples/snippets/generated"
+- "/java-websecurityscanner/grpc-google-.*/src"
+- "/java-websecurityscanner/proto-google-.*/src"
+- "/java-websecurityscanner/google-.*/src"
 
 deep-preserve-regex:
-- "/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
+- "/java-websecurityscanner/google-.*/src/test/java/com/google/cloud/.*/v.*/it/IT.*Test.java"
 - "/google-cloud-websecurityscanner/src/test/java/com/google/cloud/websecurityscanner/it/v1beta/VPCServiceControlNegativeTest.java"
 - "/google-cloud-websecurityscanner/src/test/java/com/google/cloud/websecurityscanner/it/v1beta/VPCServiceControlPositiveTest.java"
```
Comment (suztomo) [inline java-websecurityscanner/.OwlBot.yaml:25]: These are missed

File: generation/set_owlbot_config.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+dir_name=$(dirname "${OWLBOT_FILE}")
+module_name=$(basename "${dir_name}")
+
+if [ ! -d "${module_name}" ]; then
+  echo "module ${module_name} does not exist"
+  exit 1
+fi
+
+# For deep-remove-regex and deep-preserve-regex fields
+sed -i "s|\"/grpc-google|\"/${module_name}/grpc-google|" "${OWLBOT_FILE}"
```
Comment (meltsufin) [inline generation/set_owlbot_config.sh:34]: Can you use `-i.bak` to make it work on macOS please?

**Review by meltsufin — APPROVED** [review]

**Review by lqiu96 — APPROVED** [review]

File: generation/set_owlbot_config.sh (hunk 7 above)
Comment (suztomo) [reply generation/set_owlbot_config.sh:34]: I'm afraid that you said you don't ask others to adopt macOS shell.

Comment (meltsufin) [reply generation/set_owlbot_config.sh:34]: Haha ok be like that. 😄

**Pull request merged**